src/software/gui/standalone_simulator/widgets/standalone_simulator_draw_function_visualizer.cpp

```diff
@@ -41,6 +46,16 @@ void StandaloneSimulatorDrawFunctionVisualizer::mousePressEvent(QMouseEvent* eve
 void StandaloneSimulatorDrawFunctionVisualizer::mouseReleaseEvent(QMouseEvent* event)
 {
     robot.reset();
+    if (ctrl_clicked)
```

Comment:
what if we checked  `event->modifiers() & Qt::ControlModifier ` here instead? initial_point could be std::optional

Reply:
I think we should leave it as a boolean, because in mouseMoveEvent if we also remove ctrl_clicked, we wouldn't know if a user clicked ctrl and then moved the ball or just held down ctrl and moved the ball.

Reply:
Wouldn't we have to update ctrl_clicked in mouseMoveEvent in case the user releases the ctrl key while moving the mouse?

Reply:
Hmm I guess this depends on what you expect to happen when ctrl is released. Should we release the ball when ctrl is released or just release it when the mouse click is released?

Reply:
I was thinking that we only need to check when the mouse is initial clicked, so if ctrl is released before mouse is released, we still set the ball velocity. However, we should get some other opinions on this

Reply:
@akhilveeraghanta @jared-paul @nimazareian @jsn0716 thoughts?

Reply:
I like the idea of having it so that you must hold CTRL throughout the entire process. So we release the ball when the mouse click is released.

Reply:
I tried it out in sim; first of all @Resocram awesome work on the feature! it works really well

I'm indifferent to both approaches, but if I had to choose, the current approach seems pretty good (I'd recommend trying it out in sim if you haven't already), I can hold CNTRL down and do multiple shots. Or I can click control, drag my mouse, release CNTRL and spend more time orienting things before firing.

Reply:
Basically I think it's tricky handling all the cases when ctrl is released while the mouse button is pressed. I think holding ctrl throughout would also work even if we only checked ctrl when the mouse is first pressed

Reply:
> I think holding ctrl throughout would also work even if we only checked ctrl when the mouse is first pressed

That is true
 
>  if ctrl is released before mouse is released, we still set the ball velocity. 

This is the part that I _feel_ I wouldn't like as much (again super subjective, and I don't feel that strongly)

Reply:
> > if ctrl is released before mouse is released, we still set the ball velocity.
> 
> This is the part that I _feel_ I wouldn't like as much (again super subjective, and I don't feel that strongly)

what should happen in that case then?

Reply:
Only releasing the mouse should set the ball velocity, so when u release cntrl before the mouse is released it should do nothing (which is how it is rn in this PR)

Reply:
I just tried this and releasing ctrl early works for me. I'm ok with the current behaviour

Reply:
Velocity being set on mouse release seems the most intuitive to me, but don't really have a strong opinion. On a side note - do we have documentation for how to use the simulation? Features like this would be something I wouldn't want to have to learn by reading code

Reply:
> Velocity being set on mouse release seems the most intuitive to me, but don't really have a strong opinion. On a side note - do we have documentation for how to use the simulation? Features like this would be something I wouldn't want to have to learn by reading code

That is a good point, I didn't really know how to use the simulator until Jonathan showed me a demo of it. I can add it to the READme in another pr.



> Only releasing the mouse should set the ball velocity, so when u release cntrl before the mouse is released it should do nothing (which is how it is rn in this PR)

Yeah I personally like the idea of releasing the ball when the mouse is released, it seems more intuitive imo

Reply:
> Yeah I personally like the idea of releasing the ball when the mouse is released, it seems more intuitive imo

I agree with this. The question was what happens when ctrl is released before mouse is released. As it is right now, nothing happens and only mouse release causes the ball to start moving, which is fine with me but can be done with a bit less code.